web/src/main/java/org/akaza/openclinica/control/submit/DownloadAttachedFileServlet.java

```diff
@@ -70,6 +70,11 @@ public void processRequest() throws Exception {
         String filePathName = "";
         String fileName = fp.getString("fileName");
         File f = new File(fileName);
+        
+        if(fileName != null && fileName.indexOf("..") > -1) {

```

Comment:
Instead of checking for "..", use File#getCanonicalPath which will remove . and .. and resolve any symlinks. You can then check if the canonical path starts with the base dir path that is allowed.
Here is an article for reference: https://www.stackhawk.com/blog/spring-path-traversal-guide-examples-and-prevention/.

Reply:
changed--check for canonical path  and only allow if the canonical path starts with the base dir path.